Show something useful when failing to ssh into a project/environment

```
lagoon ssh -p education-sa -e master
panic: Failed to dial: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
goroutine 1 [running]:
github.com/amazeeio/lagoon-cli/lagoon/ssh.InteractiveSSH(0xc000141b80, 0x0, 0x0, 0x0, 0x0, 0xc000141bb0)
	/go/src/github.com/amazeeio/lagoon-cli/lagoon/ssh/main.go:17 +0x90a
github.com/amazeeio/lagoon-cli/cmd.glob..func48(0x1a6b280, 0xc0000def80, 0x0, 0x4)
	/go/src/github.com/amazeeio/lagoon-cli/cmd/ssh.go:58 +0x76b
github.com/spf13/cobra.(*Command).execute(0x1a6b280, 0xc0000deec0, 0x4, 0x4, 0x1a6b280, 0xc0000deec0)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0x1a6ab00, 0x0, 0x0, 0x14d0b23)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/amazeeio/lagoon-cli/cmd.Execute()
	/go/src/github.com/amazeeio/lagoon-cli/cmd/root.go:61 +0x42
main.main()
	/go/src/github.com/amazeeio/lagoon-cli/main.go:8 +0x20
```

It was that the project name was incorrect after all, but the above errors lead me to believe that there was something wrong with the authentication. It could be somethings like "no such project" or "no such environment" (suggestions of close matches would be a plus 🙂😉)

--- a/cmd/ssh.go
+++ b/cmd/ssh.go
@@ -5,6 +5,7 @@ import (
 	"os"
 
 	lagoonssh "github.com/amazeeio/lagoon-cli/lagoon/ssh"
+	"github.com/amazeeio/lagoon-cli/output"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 	"golang.org/x/crypto/ssh"
@@ -53,10 +54,14 @@ var sshEnvCmd = &cobra.Command{
 				HostKeyCallback: ssh.InsecureIgnoreHostKey(),
 			}
 			defer closeSSHAgent()
+			var err error
 			if sshCommand != "" {
-				lagoonssh.RunSSHCommand(sshConfig, sshService, sshContainer, sshCommand, config)
+				err = lagoonssh.RunSSHCommand(sshConfig, sshService, sshContainer, sshCommand, config)
 			} else {
-				lagoonssh.InteractiveSSH(sshConfig, sshService, sshContainer, config)
+				err = lagoonssh.InteractiveSSH(sshConfig, sshService, sshContainer, config)
+			}
+			if err != nil {
+				output.RenderError(err.Error(), outputOptions)
 			}
 		}
 

--- a/lagoon/ssh/main.go
+++ b/lagoon/ssh/main.go
@@ -2,8 +2,8 @@ package ssh
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
-	"log"
 	"os"
 
 	"golang.org/x/crypto/ssh"
@@ -11,16 +11,16 @@ import (
 )
 
 // InteractiveSSH .
-func InteractiveSSH(lagoon map[string]string, sshService string, sshContainer string, config *ssh.ClientConfig) {
+func InteractiveSSH(lagoon map[string]string, sshService string, sshContainer string, config *ssh.ClientConfig) error {
 	client, err := ssh.Dial("tcp", lagoon["hostname"]+":"+lagoon["port"], config)
 	if err != nil {
-		panic("Failed to dial: " + err.Error())
+		return errors.New("Failed to dial: " + err.Error() + "\nPossibly check the project or environment name")
 	}
 
 	// start the session
 	session, err := client.NewSession()
 	if err != nil {
-		panic("Failed to create session: " + err.Error())
+		return errors.New("Failed to create session: " + err.Error())
 	}
 	defer session.Close()
 	session.Stdout = os.Stdout
@@ -35,16 +35,16 @@ func InteractiveSSH(lagoon map[string]string, sshService string, sshContainer st
 	if terminal.IsTerminal(fileDescriptor) {
 		originalState, err := terminal.MakeRaw(fileDescriptor)
 		if err != nil {
-			log.Fatalf("%s", err)
+			return err
 		}
 		defer terminal.Restore(fileDescriptor, originalState)
 		termWidth, termHeight, err := terminal.GetSize(fileDescriptor)
 		if err != nil {
-			log.Fatalf("%s", err)
+			return err
 		}
 		err = session.RequestPty("xterm-256color", termHeight, termWidth, modes)
 		if err != nil {
-			log.Fatalf("%s", err)
+			return err
 		}
 	}
 	var connString string
@@ -56,22 +56,23 @@ func InteractiveSSH(lagoon map[string]string, sshService string, sshContainer st
 	}
 	err = session.Start(connString)
 	if err != nil {
-		log.Fatalf("failed to start shell: %s", err)
+		return errors.New("Failed to start shell: " + err.Error())
 	}
 	session.Wait()
+	return nil
 }
 
 // RunSSHCommand .
-func RunSSHCommand(lagoon map[string]string, sshService string, sshContainer string, command string, config *ssh.ClientConfig) {
+func RunSSHCommand(lagoon map[string]string, sshService string, sshContainer string, command string, config *ssh.ClientConfig) error {
 	client, err := ssh.Dial("tcp", lagoon["hostname"]+":"+lagoon["port"], config)
 	if err != nil {
-		panic("Failed to dial: " + err.Error())
+		return errors.New("Failed to dial: " + err.Error() + "\nPossibly check the project or environment name")
 	}
 
 	// start the session
 	session, err := client.NewSession()
 	if err != nil {
-		panic("Failed to create session: " + err.Error())
+		return errors.New("Failed to create session: " + err.Error())
 	}
 	defer session.Close()
 	session.Stdout = os.Stdout
@@ -86,16 +87,16 @@ func RunSSHCommand(lagoon map[string]string, sshService string, sshContainer str
 	if terminal.IsTerminal(fileDescriptor) {
 		originalState, err := terminal.MakeRaw(fileDescriptor)
 		if err != nil {
-			log.Fatalf("%s", err)
+			return err
 		}
 		defer terminal.Restore(fileDescriptor, originalState)
 		termWidth, termHeight, err := terminal.GetSize(fileDescriptor)
 		if err != nil {
-			log.Fatalf("%s", err)
+			return err
 		}
 		err = session.RequestPty("xterm-256color", termHeight, termWidth, modes)
 		if err != nil {
-			log.Fatalf("%s", err)
+			return err
 		}
 	}
 	var connString string
@@ -114,6 +115,7 @@ func RunSSHCommand(lagoon map[string]string, sshService string, sshContainer str
 		os.Exit(1)
 	}
 	fmt.Println(b.String())
+	return nil
 }
 
 // GenerateSSHConnectionString .

--- a/lagoon/ssh/main.go
+++ b/lagoon/ssh/main.go
@@ -14,7 +14,7 @@ import (
 func InteractiveSSH(lagoon map[string]string, sshService string, sshContainer string, config *ssh.ClientConfig) error {
 	client, err := ssh.Dial("tcp", lagoon["hostname"]+":"+lagoon["port"], config)
 	if err != nil {
-		return errors.New("Failed to dial: " + err.Error() + "\nPossibly check the project or environment name")
+		return errors.New("Failed to dial: " + err.Error() + "\nCheck that the project or environment you are trying to connect to exists")
 	}
 
 	// start the session
@@ -66,7 +66,7 @@ func InteractiveSSH(lagoon map[string]string, sshService string, sshContainer st
 func RunSSHCommand(lagoon map[string]string, sshService string, sshContainer string, command string, config *ssh.ClientConfig) error {
 	client, err := ssh.Dial("tcp", lagoon["hostname"]+":"+lagoon["port"], config)
 	if err != nil {
-		return errors.New("Failed to dial: " + err.Error() + "\nPossibly check the project or environment name")
+		return errors.New("Failed to dial: " + err.Error() + "\nCheck that the project or environment you are trying to connect to exists")
 	}
 
 	// start the session

